controlplane/src/core/repositories/OrganizationRepository.ts

```diff
@@ -597,4 +606,250 @@ export class OrganizationRepository {
+              type: 'slack',
+              events: input.events,
+            })
+            .returning()
+            .execute();
+
+          const slackIntegrationConfig = await db
+            .insert(slackIntegrationConfigs)
+            .values({
+              integrationId: createSlackIntegrationResult[0].id,
+              endpoint: input.endpoint,
+            })
+            .returning()
+            .execute();
+
+          for (const eventMeta of input.eventsMeta) {
+            switch (eventMeta.meta.case) {
+              case 'federatedGraphSchemaUpdated': {
+                const ids = eventMeta.meta.value.graphIds;
+                await db.insert(slackSchemaUpdateEventConfigs).values(
+                  ids.map((id) => ({
+                    slackIntegrationConfigId: slackIntegrationConfig[0].id,
+                    federatedGraphId: id,
+                  })),
+                );
+                break;
+              }
+              default: {
+                throw new Error(`This event ${eventMeta.meta.case} doesnt exist`);
+              }
+            }
+          }
+        }
+      }
+    });
+  }
+
+  public async getIntegrationByName(organizationId: string, integrationName: string): Promise<Integration | undefined> {
+    const res = await this.db.query.organizationIntegrations.findFirst({
+      where: and(
+        eq(organizationIntegrations.organizationId, organizationId),
+        eq(organizationIntegrations.name, integrationName),
+      ),
+    });
+
+    if (!res) {
+      return undefined;
+    }
+
+    switch (res.type) {
+      case integrationTypeEnum.enumValues[0]: {
+        const slackIntegrationConfig = await this.db.query.slackIntegrationConfigs.findFirst({
+          where: eq(slackIntegrationConfigs.integrationId, res.id),
+          with: {
+            slackSchemUpdateEventConfigs: true,
+          },
+        });
+
+        if (!slackIntegrationConfig) {
+          return undefined;
+        }
+
+        const config: PartialMessage<IntegrationConfig> = {
+          type: IntegrationType.SLACK,
+          config: {
+            case: 'slackIntegrationConfig',
+            value: {
+              endpoint: slackIntegrationConfig.endpoint,
+            },
+          },
+        };
+
+        return {
+          id: res.id,
+          name: res.name,
+          type: res.type,
+          events: res.events || [],
+          integrationConfig: config,
+          eventsMeta: [
+            {
+              eventName: OrganizationEventName.FEDERATED_GRAPH_SCHEMA_UPDATED,
+              meta: {
+                case: 'federatedGraphSchemaUpdated',
+                value: {
+                  graphIds: slackIntegrationConfig.slackSchemUpdateEventConfigs.map((i) => i.federatedGraphId),
+                },
+              },
+            },
+          ],
+        } as Integration;
+      }
+      default: {
+        throw new Error('The type of the integration doesnt exist');
```

Comment:
Specify in the error what value was wrong.

Reply:
Done